pipeline_dp/dataset_histograms/histograms.py

```diff
@@ -27,24 +27,38 @@ class FrequencyBin:
 
     Attributes:
       lower: the lower bound of the bin.
+      upper: the upper of the bin, i.e. values in the bin are in
+        [lower; upper} range ("}" means it can be either "]" or ")", see
+        upper_included to understand when it is "]" and when ")").
+      upper_included: if true, then bin range is [lower; upper], otherwise
+        [lower; upper). It can be included only for the last bin.
       count: the number of elements in the bin.
       sum: the sum of elements in the bin.
       max: the maximum element in the bin, which is smaller or equal to the
-       upper-1.
+        upper.
     """
     lower: Union[int, float]
+    upper: Union[int, float]
+    upper_included: bool
```

Comment:
`upper_included` feels not very neccessary, it's set to true only on the last bin in sum histogram.
WDYT about the following:
1. Put `lower`/`upper` fields in `Histogram` (which seems to be useful independently of this)
2. When bin.upper == histogram.upper it would automatically mean that upper_included

Of course another option is to assume that `upper_included = False` always, and to create additional bin which would include upper bound of `Histogram` (with `bin.lower = Histogram.upper`,` bin.upper = bin.lower + small_number`)

WDYT?

Reply:
I took a look and I think putting into histogram lower/upper will require quite some refactoring as we will need to propagate these values in PCollections. However, I think we can even avoid it at all because we will know the bounds always:
1. for int histogram we just take the last bin for upper and 1 as lower, as we don't have clear range in the histogram, it is kind of [1; +inf)
2. for float (sum) histogram we take first and last bins, they will be always present and then their lower and upper will define the histogram range (and only in this case for the last bin of the float hist the upper will be included)

And in the end, looks like we need upper inside bins more and for the histogram as a whole we don't really need it as for now and in general we don't care much whether the value is included or not, because when I removed now upper_included no tests failed.

WDYT?
Take a look, I left some comments in the code that explain that.